### PR TITLE
First Prototype

### PR DESCRIPTION
**Pull request by TomasJavurek**
first prototype

### CODE REVIEW

**Comment by bari12** [comment]
Please push to your private repositories and even work on the branches together there. There is no reason to push the branches to the main repository for communication.

**Comment by TomasJavurek** [comment]
I would like to push the new version to the master. Look at #640

**Comment by bari12** [comment]
> lib/rucio/client/downloadclient.py:13:11: E261 at least two spaces before inline comment
> lib/rucio/client/downloadclient.py:13:12: E262 inline comment should start with '# '
> lib/rucio/client/downloadclient.py:23:1: F401 'rucio.common.exception.SourceAccessDenied' imported but unused
> lib/rucio/client/downloadclient.py:24:1: F401 'rucio.common.exception.SourceNotFound' imported but unused
> lib/rucio/client/downloadclient.py:25:1: F401 'rucio.common.exception.RSEAccessDenied' imported but unused
> lib/rucio/client/downloadclient.py:26:1: F401 'rucio.common.exception.FullStorage' imported but unused
> lib/rucio/client/downloadclient.py:33:1: F401 'threading.Event' imported but unused
> lib/rucio/client/downloadclient.py:37:7: E225 missing whitespace around operator
> lib/rucio/client/downloadclient.py:39:1: E265 block comment should start with '# '
> lib/rucio/client/downloadclient.py:47:1: E302 expected 2 blank lines, found 1
> lib/rucio/client/downloadclient.py:63:5: E303 too many blank lines (3)
> lib/rucio/client/downloadclient.py:63:140: W291 trailing whitespace
> lib/rucio/client/downloadclient.py:65:9: E265 block comment should start with '# '
> lib/rucio/client/downloadclient.py:75:30: E231 missing whitespace after ':'
> lib/rucio/client/downloadclient.py:94:13: E265 block comment should start with '# '
> lib/rucio/client/downloadclient.py:151:17: E303 too many blank lines (2)
> lib/rucio/client/downloadclient.py:161:38: E128 continuation line under-indented for visual indent
> lib/rucio/client/downloadclient.py:162:38: E128 continuation line under-indented for visual indent
> lib/rucio/client/downloadclient.py:163:38: E128 continuation line under-indented for visual indent
> lib/rucio/client/downloadclient.py:164:38: E128 continuation line under-indented for visual indent
> lib/rucio/client/downloadclient.py:165:38: E128 continuation line under-indented for visual indent
> lib/rucio/client/downloadclient.py:168:13: E303 too many blank lines (2)
> lib/rucio/client/downloadclient.py:176:54: E128 continuation line under-indented for visual indent
> lib/rucio/client/downloadclient.py:177:54: E128 continuation line under-indented for visual indent
> lib/rucio/client/downloadclient.py:178:54: E128 continuation line under-indented for visual indent
> lib/rucio/client/downloadclient.py:179:54: E128 continuation line under-indented for visual indent
> lib/rucio/client/downloadclient.py:180:54: E128 continuation line under-indented for visual indent
> lib/rucio/client/downloadclient.py:181:53: E261 at least two spaces before inline comment
> lib/rucio/client/downloadclient.py:244:29: E265 block comment should start with '# '
> lib/rucio/client/downloadclient.py:252:5: E303 too many blank lines (2)
> lib/rucio/client/downloadclient.py:256:40: E261 at least two spaces before inline comment
> lib/rucio/client/downloadclient.py:267:34: E261 at least two spaces before inline comment
> lib/rucio/client/downloadclient.py:267:35: E262 inline comment should start with '# '
> lib/rucio/client/downloadclient.py:268:23: E265 block comment should start with '# '
> lib/rucio/client/downloadclient.py:269:23: E265 block comment should start with '# '
> lib/rucio/client/downloadclient.py:270:23: E265 block comment should start with '# '
> lib/rucio/client/downloadclient.py:271:23: E265 block comment should start with '# '
> lib/rucio/client/downloadclient.py:299:5: E303 too many blank lines (2)
> lib/rucio/client/downloadclient.py:320:16: E111 indentation is not a multiple of four
> lib/rucio/client/downloadclient.py:348:28: F821 undefined name 'RSENotFound'
> lib/rucio/client/downloadclient.py:368:24: F821 undefined name 'RSEProtocolNotSupported'
> lib/rucio/client/downloadclient.py:414:51: F821 undefined name 'signal'
> lib/rucio/client/downloadclient.py:455:5: E303 too many blank lines (2)
> lib/rucio/client/downloadclient.py:459:23: F821 undefined name 'RucioException'
> lib/rucio/client/downloadclient.py:473:5: E303 too many blank lines (2)
> lib/rucio/client/downloadclient.py:490:5: E303 too many blank lines (2)
> lib/rucio/client/downloadclient.py:511:1: W391 blank line at end of file
😄

**Comment by TomasJavurek** [comment]
well this was the old version.

**Comment by bari12** [comment]
Then the new one wasn't pushed to git. Just checking a few of these errors, they are all still valid.

**Comment by bbockelm** [comment]
Yeah - please don't push to the main repo.

Additionally, you might find these links helpful: 

* https://github.com/rucio/rucio/blob/master/CONTRIBUTING.rst
* https://wiki.openstack.org/wiki/GitCommitMessages#Information_in_commit_messages (@bari12 - in fact, that might be a good one to link from the contributor's guide).

File: lib/rucio/client/downloadclient.py [hunk 1]
```diff
@@ -0,0 +1,483 @@
...
+FAILURE = 1
+
+DEFAULT_SECURE_PORT = 443
+DEFAULT_PORT = 80
+
+
+class DownloadClient(BaseClient):
+    """ This class cover all functionality related to file uploads into Rucio."""
+
+    BASEURL = '??_downloadclient_??'
```
Comment (bari12) [inline lib/rucio/client/downloadclient.py:48]: Remove

**Review by bari12 — COMMENTED** [review]
There are some pylint remarks which should be adressed eventually. But the score (7.6) is not too bad.
I know you want to test this on the grid and as it is not included anywhere I think we can merge this; For the further development I would propose that you two do this on a private branch, so we don't have to go through the code iterations on the main repository.

**Review by bari12 — APPROVED** [review]

**Pull request merged**